src/index.js

```diff
@@ -303,6 +304,8 @@ class FloodSub extends BaseProtocol {
     function checkIfReady (peer) {
       if (peer && peer.isWritable) {
         peer.sendUnsubscriptions(topics)
+      } else {
+        process.nextTick(checkIfReady.bind(peer))
```

Comment:
We used the `setImmediate` from the `async` module in the past to avoid the browser transpilation pitfalls. Is this `process` module an improvement? It seems it hasn't been maintainer for 3+ years.

Review https://github.com/defunctzombie/node-process/issues & https://github.com/defunctzombie/node-process/pulls to check if there is anything that will bite us.

Reply:
Ah, even the fabulous @satazor hit this one before https://github.com/webpack/node-libs-browser/issues/92 and it seems to recommend using a `setImmediate`

Reply:
When we had the nearform team working on IPFS benchmarking, we also moved away to `setImmediate` in favor of `process.nextTick`, which is already supported by node and we have been using it in several modules:

https://nodejs.org/api/process.html#process_process_nexttick_callback_args

Reply:
Did you confirm it works with Browserify? In the past, we moved away from the native .nexttick because browserify would not polyfill it correctly (or at all)

Reply:
Note: It is always recommended to verify what are the default polyfills on each blunder for each of Node.js core modules. Otherwise you might be injecting a slow or broken code path that the Node.js tests won't catch. (Happened many times before)

Reply:
That's true! I switched to [async/nextTick](https://caolan.github.io/async/v3/nextTick.js.html), as we are doing in the other places. It takes care of other environments